Microsoft.Toolkit.Uwp/Helpers/BackgroundTaskHelper.cs

```diff
@@ -0,0 +1,107 @@
+        /// <param name="backgroundTaskType">The type of the background task.</param>
+        /// <returns>True/False indicating if a background task was registered or not</returns>
+        public static bool IsBackgroundTaskRegistered(Type backgroundTaskType)
+        {
+            return BackgroundTaskRegistration.AllTasks.Any(t => t.Value.Name == backgroundTaskType.Name);
+        }
+
+        /// <summary>
+        /// Register a background task for the application with conditions
+        /// </summary>
+        /// <param name="backgroundTaskType">The type of the background task</param>
+        /// <param name="trigger">Trigger that indicate when the background task should be invoked</param>
+        /// <param name="enforceConditions">Indicate if the background task should quit if condition is no longer valid</param>
+        /// <param name="conditions">Optional conditions for the background task to run with</param>
+        /// <returns>Background Task that was registered with the system</returns>
+        public static async Task<BackgroundTaskRegistration> RegisterAsync(Type backgroundTaskType, IBackgroundTrigger trigger, bool enforceConditions = true, params IBackgroundCondition[] conditions)
+        {
+            // Check if the task is already registered.
+            if (IsBackgroundTaskRegistered(backgroundTaskType) == true)
+            {
+                return GetBackgroundTask(backgroundTaskType) as BackgroundTaskRegistration;
+            }
+
+            // Verify access
+            BackgroundAccessStatus taskRequired = await BackgroundExecutionManager.RequestAccessAsync();
+
+            if (taskRequired == BackgroundAccessStatus.Denied)
+            {
+                throw new InvalidOperationException("Background access is denied!");
+            }
+
+            // Get details about the background task
+            string backgroundTaskName = backgroundTaskType.Name;
+            string backgroundTaskEntryPoint = backgroundTaskType.FullName;
+
+            // build the background task
+            BackgroundTaskBuilder builder = new BackgroundTaskBuilder();
+            builder.Name = backgroundTaskName;
+            builder.TaskEntryPoint = backgroundTaskEntryPoint;
+            builder.CancelOnConditionLoss = enforceConditions;
+
+            // add the conditions if we have them.
+            conditions?.ToList().ForEach(builder.AddCondition);
+
+            builder.SetTrigger(trigger);
+
+            // Register it
+            BackgroundTaskRegistration registered = builder.Register();
+
+            return registered;
+        }
+
+        /// <summary>
+        /// Unregister a background task
+        /// </summary>
+        /// <param name="backgroundTaskType">The type of the background task</param>
+        public static void Unregister(Type backgroundTaskType)
+        {
+            if (IsBackgroundTaskRegistered(backgroundTaskType) == false)
+            {
+                return;
+            }
+
```

Comment:
Unnecessary to check if registered - just call GetBackgroundTask, and if it's not null, unregister it, right?